### PR TITLE
python3Packages.bucketstore: init at 0.2.2

### PR DESCRIPTION
**Pull request by jpetrucciani**
###### Description of changes

I would like to be able to use `bucketstore` from nixpkgs proper instead of adding overlays to include it in my various python repos!

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/bucketstore/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  version = "0.2.2";
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "00xl1wjpv7417pnsxml22ibkjxakqv96x7gpq2jya3fv7viwp1sd";
+  };
+
+  propagatedBuildInputs = [ boto3 ];
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/bucketstore/default.nix:19]: The repo contains [tests](https://github.com/jpetrucciani/bucketstore/tree/master/tests). Using from `fetchFromGitHub` would perhaps allow to execute them if they don't need network access. A short comment that stated why the tests are disabled would help for the future.

File: pkgs/development/python-modules/bucketstore/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pythonImportsCheck = [
+    "bucketstore"
+  ];
+
+  meta = with lib; {
+    description = "A simple library for interacting with Amazon S3.";
```
Comment (fabaff) [inline pkgs/development/python-modules/bucketstore/default.nix:25]:
```suggestion
    description = "Library for interacting with Amazon S3";
```

File: pkgs/development/python-modules/bucketstore/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    homepage = "https://github.com/jpetrucciani/bucketstore.git";
```
Comment (fabaff) [inline pkgs/development/python-modules/bucketstore/default.nix:26]:
```suggestion
    homepage = "https://github.com/jpetrucciani/bucketstore";
```

Comment (jpetrucciani) [reply pkgs/development/python-modules/bucketstore/default.nix]: good catch! updated

**Comment by jpetrucciani** [comment]
@fabaff Thanks for the review! 😄

File: pkgs/development/python-modules/bucketstore/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    repo = "bucketstore";
+    rev = version;
+    sha256 = "sha256-BtoyGqFbeBhGQeXnmeSfiuJLZtXFrK26WO0SDlAtKG4=";
+  };
+
+  checkInputs = [
+    pytestCheckHook
+    moto
+  ];
+  propagatedBuildInputs = [ boto3 ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/bucketstore/default.nix:26]:
```suggestion
  propagatedBuildInputs = [ boto3 ];

  checkInputs = [
    pytestCheckHook
    moto
  ];
```

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.bucketstore

**Pull request merged**